### PR TITLE
chore: add Github example

### PR DESCRIPTION
**Pull request by frixaco**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
      
Thank you for the library!
I didn't find any Github OAuth example and thought this might be a good chance to contribute to library that I'm using. So I added one.
Let me know if this is needed or if you have any type of feedback!

~~Waiting for #174 to be answered.~~

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
I'm not sure you'd get an answer for #174. If you'd like to send a PR for that it would be awesome.

**Comment by frixaco** [comment]
@mcollina oh ok, so it's like a bug or something? I could do a PR for that and then remove the comment from this PR for updating `refresh_token`

File: examples/github.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+let refreshToken = ''
+
+fastify.get('/login/github/callback', function (request, reply) {
+  this.githubOAuth2.getAccessTokenFromAuthorizationCodeFlow(
+    request,
+    (err, result) => {
+      if (err) {
+        reply.send(err)
+        return
```
Comment (Fdawgs) [inline examples/github.js:31]: Don't think the returns are needed after the reply.send()s.

**Comment by mcollina** [comment]
> oh ok, so it's like a bug or something? 

I have no real clue, you have likely more experience than me using this module with GitHub ;).

File: examples/github.js (hunk 1 above)
Comment (Eomm) [reply examples/github.js:31]: it is used to do not execute the `sget.concat`

File: examples/github.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      }
+      reply.send(data)
+    }
+  )
+})
+
+fastify.get('/login/github/refreshToken', function (request, reply) {
+  // this.githubOAuth2.getAccessTokenFromRefreshToken(refresh_token, {}, (err, result) => {})
+
+  sget.concat(
```
Comment (climba03003) [inline examples/github.js:86]: I would not use `sget` here.
See my comment in https://github.com/fastify/fastify-oauth2/issues/174#issuecomment-1220414854

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: examples/github.js (hunk 2 above)
Comment (frixaco) [reply examples/github.js]: Thanks @climba03003

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**